Android/app/src/main/java/io/github/project_travel_mate/login/LoginPresenter.java

```diff
@@ -172,5 +189,39 @@ public void run() {
             }
         });
     }
+    public void validateSignUp(final String firstName,final String lastName, final String email, String pass) throws ValidationException {
+        if(firstName.trim().equals("")){
```

Comment:
add firstName == null check also. This will crash when firstName is null.

Also, this check is being used again & again. You can add a new private function here that just checks if the input string is not null & not just spaces.